In [ ]:
# %pip install google-adk

  Using cached uritemplate-4.2.0-py3-none-any.whl.metadata (2.6 kB)
  Using cached proto_plus-1.26.1-py3-none-any.whl.metadata (2.2 kB)
  Using cached pyparsing-3.2.5-py3-none-any.whl.metadata (5.0 kB)
INFO: pip is looking at multiple versions of opentelemetry-exporter-otlp-proto-http to determine which version is compatible with other requirements. This could take a while.
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.2/2.2 MB 10.6 MB/s  0:00:00 eta 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.6/14.6 MB 11.2 MB/s  0:00:01m0:00:0100:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.1/8.1 MB 11.3 MB/s  0:00:00 eta 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 10.8 MB/s  0:00:00 eta 0:00:01
Using cached proto_plus-1.26.1-py3-none-any.whl (50 kB)
Using cached pyparsing-3.2.5-py3-none-any.whl (113 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 10.6 MB/s  0:00:00
Using cached uritemplate-4.2.0-py3-none-any.whl (11 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

In [ ]:
# %pip install python-dotenv

Note: you may need to restart the kernel to use updated packages.


In [15]:
from dotenv import load_dotenv
import os
from pathlib import Path

env_path = Path('.') / '.env'
load_dotenv(dotenv_path=env_path)

# reference: https://stackoverflow.com/questions/16924471/difference-between-os-getenv-and-os-environ-get
try:
    api_key = os.environ["GOOGLE_API_KEY"]
    os.environ["GOOGLE_GENAI_USE_VERTEXAI"] = "FALSE"
    print("✅ Gemini API key setup complete.")
except Exception as e:
    print(f"🔑 Authentication Error: Please make sure you have added 'GOOGLE_API_KEY' in .env file. Details: {e}")

✅ Gemini API key setup complete.


In [8]:
from google.adk.agents import Agent
from google.adk.runners import InMemoryRunner
from google.adk.tools import google_search
from google.genai import types

print("✅ ADK components imported successfully.")

✅ ADK components imported successfully.


https://google.github.io/adk-docs/agents/

- name and description: A simple name and description to identify our agent.
- model: The specific LLM that will power the agent's reasoning. We'll use "gemini-2.5-flash-lite".
- instruction: The agent's guiding prompt. This tells the agent its goal is and how to behave.
- tools: A list of tools that the agent can use. To start, we'll give it the google_search tool, which lets it find up-to-date information online.

In [9]:
root_agent = Agent(
    name="helpful_assistant",
    model="gemini-2.5-flash-lite",
    description="A simple agent that can answer general questions using Google Search.",
    instruction="You are a helpful assistant. Use the tools at your disposal to answer user queries" " accurately and concisely.",
    tools=[google_search],
)

print("✅ Root Agent defined.")

✅ Root Agent defined.


## Run my agent

In [ ]:
runner = InMemoryRunner(agent=root_agent)
print("✅ InMemoryRunner initialized."). # https://google.github.io/adk-docs/runtime/

✅ InMemoryRunner initialized.


In [11]:
ai_response = await runner.run_debug("WHat is Agent Developemnt Kit from Google? What languages is the SDK available in?" )


 ### Created new session: debug_session_id

User > WHat is Agent Developemnt Kit from Google? What languages is the SDK available in?
helpful_assistant > The Google Agent Development Kit (ADK) is an open-source, flexible, and modular framework designed to simplify the development, deployment, and orchestration of AI agents and multi-agent systems. It aims to make agent development more akin to traditional software development, offering precise control over agent behavior and orchestration. The ADK is optimized for Google's Gemini models and ecosystem but is model-agnostic and deployment-agnostic, allowing compatibility with other frameworks.

The ADK SDK is currently available in the following programming languages:
*   **Python**
*   **Go**
*   **Java**



 ### Created new session: debug_session_id

User > WHat is Agent Developemnt Kit from Google? What languages is the SDK available in?


helpful_assistant > The Google Agent Development Kit (ADK) is an open-source, flexible, and modular framework designed to simplify the development, deployment, and orchestration of AI agents and multi-agent systems. It aims to make agent development more akin to traditional software development, offering precise control over agent behavior and orchestration. The ADK is optimized for Google's Gemini models and ecosystem but is model-agnostic and deployment-agnostic, allowing compatibility with other frameworks.

The ADK SDK is currently available in the following programming languages:
*   **Python**
*   **Go**
*   **Java**

In [13]:
ai_response = await runner.run_debug("What is the currency exchange rate between EUR and KRW? And what is causing this trend or difference from yesterday's?" )


 ### Continue session: debug_session_id

User > What is the currency exchange rate between EUR and KRW? And what is causing this trend or difference from yesterday's?
helpful_assistant > As of November 11, 2025, the exchange rate is approximately 1 Euro (EUR) to 1695.85 South Korean Won (KRW).

The EUR to KRW exchange rate has seen a decrease of approximately -0.29% from the previous market day. However, over the past week, the Euro has shown a stable performance with an increase of about 1.918% compared to its value seven days ago. It's worth noting that within the past week, the rate has fluctuated, reaching a high of 1695.73 on November 11, 2025, and a low of 1645.47 on November 3, 2025. The largest 24-hour movement was a 0.321% increase on November 11, 2025.

Several factors influence the EUR to KRW exchange rate. These include global economic conditions, interest rate differentials between the Eurozone and South Korea, trade balances, and investor sentiment. The strength of the U


 ### Continue session: debug_session_id

User > What is the currency exchange rate between EUR and KRW? And what is causing this trend or difference from yesterday's?

helpful_assistant > As of November 11, 2025, the exchange rate is approximately 1 Euro (EUR) to 1695.85 South Korean Won (KRW).

The EUR to KRW exchange rate has seen a decrease of approximately -0.29% from the previous market day. However, over the past week, the Euro has shown a stable performance with an increase of about 1.918% compared to its value seven days ago. It's worth noting that within the past week, the rate has fluctuated, reaching a high of 1695.73 on November 11, 2025, and a low of 1645.47 on November 3, 2025. The largest 24-hour movement was a 0.321% increase on November 11, 2025.

Several factors influence the EUR to KRW exchange rate. These include global economic conditions, interest rate differentials between the Eurozone and South Korea, trade balances, and investor sentiment. The strength of the US dollar can also play a role.

More specifically, recent trends in the Euro's strength have been influenced by improved economic growth prospects in the Eurozone and signs of stable inflation. Conversely, the South Korean Won may be affected by its sensitivity to global investor sentiment and potential shifts in trade performance. In general, interest rate differentials, risk perception, and the performance of the Korean equity market are significant drivers of the KRW's movement against the USD.


## ADK Web Interface

In [14]:
!adk create sample-agent --model gemini-2.5-flash-lite --api_key $GOOGLE_API_KEY


Agent created in /Users/yujinchung/FIO/5days/sample-agent:
- .env
- __init__.py
- agent.py



In [17]:
!adk web --port 4200

/Users/yujinchung/miniconda3/envs/llama311/lib/python3.11/site-packages/google/adk/cli/fast_api.py:130: UserWarning: [EXPERIMENTAL] InMemoryCredentialService: This feature is experimental and may change or be removed in future versions without notice. It may introduce breaking changes at any time.
  credential_service = InMemoryCredentialService()
/Users/yujinchung/miniconda3/envs/llama311/lib/python3.11/site-packages/google/adk/auth/credential_service/in_memory_credential_service.py:33: UserWarning: [EXPERIMENTAL] BaseCredentialService: This feature is experimental and may change or be removed in future versions without notice. It may introduce breaking changes at any time.
  super().__init__()
INFO:     Started server process [15773]
INFO:     Waiting for application startup.

+-----------------------------------------------------------------------------+
| ADK Web Server started                                                      |
|                                                 